pkg/config/migrate/migrate.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+// Contains all the bits and pieces we need to parse and manage
+// the external configuration
+package migrate
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/legacy"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/config"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+const (
+	metricArg      = "metrics"
+	inventoryArg   = "inventory"
+	eventsArg      = "events"
+	exeSuffix      = ".exe"
+	prefixArg      = "--"
+	prefixArgShort = "-"
+)
+
+// V3toV4Result represents the result of a migration
+type V3toV4Result struct {
+	V3toV4Result string `json:"migrateV3toV4Result"`
+}
+
+func V3toV4(pathConfiguration string, pathDefinition string, pathOutput string, overwrite bool) string {
```

Comment:
what do you think if this returns an error in case of error? that way if you want to call it multiple times for different conf files, one error will not stop the execution

Reply:
As it is written now, you can only pass 1 config pair at a time. So if there is an error, it will fail and you will retry.

Reply:
done